confirm-deposit.md

```diff
@@ -0,0 +1,109 @@
+
+## One-Time Setup
+
+1. Install [node.js](https://nodejs.org/en/download/). 
+On the command line, run `node -v`  to check your installation.
+
+2. Clone the Flare repository: 
+
+`git clone https://github.com/flare-foundation/flare.git
+cd flare`
+
+See [the Flare repository on GitHub](https://github.com/flare-foundation/flare).
+
+3. Install the web3.js library: 
+
+`npm install web3`
+
+See [Adding web3.js](https://web3js.readthedocs.io).
+
+4. Build Flare with this script:
+
+`./scripts/build.sh`
+
+## Run the Code
+
+1. `cd` to `flare`
+
+2. Run the code: 
+
+The following code:
+
+* Sets up the deposit by instantiating `web3`, assigning the endpoint URL, and assigning the `receivingAddress`
+* Subscribes to 'pending transactions` to listen for deposit transactions and, if they are sent to the receiving address, labels them as "pending."
+* Subscribes to 'newBlockHeaders` to get a block that is five blocks back from the block with the pending transactiions and labels it as "confirmed."
+
+```javascript
+// You must have web3 installed 
+const Web3 = require('web3');
+
+// Test this code on the Coston testnet. Use your own node URL for actual runtime.
+// https://docs.flare.network/dev/reference/coston-testnet/
+const web3 = new Web3("wss://coston-api.flare.network/ext/bc/C/ws");
+
+// This is the testnet receiving address. Use your receiving wallet address for actual runtime.
+const receivingAddress = "0x947c76694491d3fD67a73688003c4d36C8780A97";
+
+// Monitor for newly submitted deposit transactions.
+web3.eth.subscribe("pendingTransactions")
+// When a new transaction hash is received...
+.on("data", async (transactionHash) => {
+    // retrieve the transaction.
+    let tx = await web3.eth.getTransaction(transactionHash);
+    // If it is directed to our address...
+    if (tx.to === receivingAddress) {
+        // mark it as pending.
+        console.log("Transaction", tx.hash, "is pending");
+    }
+}).on("error", console.error);
+
+// Retrieve blocks 5 blocks back for confirmation.
+web3.eth.subscribe("newBlockHeaders")
+// When a new block has been produced...
+.on("data", async (blockHeader) => {
+    // retrieve a block old enough to be considered confirmed.
+    let block = await web3.eth.getBlock(blockHeader.number - 5);
+
+    // Get all of its transactions.
+    block.transactions.forEach(async (transactionHash) => {
+        // Retrieve the transaction...
+        let tx = await web3.eth.getTransaction(transactionHash);
+        // If it is directed to our address...
+        if (tx.to === receivingAddress) {
+            // mark it as confirmed.
+            console.log("Transaction", tx.hash,
+                "is confirmed in block", block.number);
+        }
+    });
+}).on("error", console.error);
+```
+
+### Outcome
+
+You can see which transactions are pending and which are confirmed and the block number they are in.
+
+For example,
+
+```javascript
```

Comment:
Well, this code block is not javascript. But we're not strict about this, this is only used to produce syntax coloring so you are free to use whatever "looks good".

Reply:
Obvioulsy not a language I know. What is it? Typescript?

Reply:
It's isn't any language, just console output. So the language tag should be just `text`, meaning that no syntax highlighting will be used.
That said, sometimes, if the console output _resembles_ some other language (like js), it's worth using `javascript` instead of `text` so you get the numbers colored differently, for example, which is nice.

Reply:
Got it. So I:
* changed mention of the code to "code," instead of "javascript."
* Gave the file suggestion as <name-of-file>.txt (It runs fine.)

Reply:
Oh, we didn't understand each other! The code starting in line 61 is definitely javascript!
What is not javascript is the console output starting in line 120, that's what my initial comment referred to.
So you can rename the file to `.js`. `.txt` will work, but it's misleading.

Reply:
👍